lib/std/os.zig

```diff
@@ -660,8 +660,35 @@ pub fn kill(pid: pid_t, sig: u8) KillError!void {
     }
 }
 
-/// Exits the program cleanly with the specified status code.
-pub fn exit(status: u8) noreturn {
+/// The type of the value that is passed to the operating system when exiting.
+/// See: `exit`.
+pub const ExitStatus = switch (builtin.os.tag) {
+    .plan9 => [:0]const u8,
+    .uefi => std.os.uefi.Status,
+    // TODO: `.wasi => u7,`
+    //       This involves updating all `main`s that use `u8` to use std.os.ExitStatus.
+    //       Also, wasmtime says the actual range is "[0..126)". Use `@Int(0, 125)` when possible.
```

Comment:
CC @Luukdegram you might have an opinion on this. Can you confirm that in WASI the maximum exit code integer is 125? wasmtime gives me an error every time I use anything above 126.

Reply:
There is no `@Int(0, 125)` yet and its likely that the underlying storage type is necessary. Do you mean std.meta.Integer ?

Reply:
That's a limitation put up by Wasmtime. Exit code is a 32-bit integer (in preview1) as defined by the specification: https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/docs.md#-exitcode-u32

Note that there is a WASI-CLI proposal that will have exit codes use [0, 255]:
https://github.com/WebAssembly/wasi-cli/issues/11

I think it's better to prepare for preview2 and use an `u8`, as there's no solution for exit code ambiguity in preview1 yet.

Reply:
@Luukdegram I've also considered using `std.os.wasi.exitcode_t` which would allow returning the full ranges of possible exit codes. But if eventually the range is basically just going to be [0, 255] like on other platforms and "256 and up = unrepresentable (Unix)", it seems good to just continue using `u8`. Thanks for your input! I'll just remove that TODO.

@matu3ba I'm aware there is no `@Int` yet, that's why I wrote "when possible". As for "its likely that the underlying storage type is necessary", that shouldn't be necessary. We should be allowed to use `@Int` here once that exists.